File: js/euler85.js
function euler85(){
    let area = 0, bestDiff = 2_000_000;
    for (let length = 1; length < 100; length++){
        for (let width = 1; width < 100; width++){
            let count = 0;
            for (let i = 1; i <= length; i++){
                for (let j = 1; j <= width; j++){
                    // number of rectangles of dimensions i*j
                    // than can fit in grid with dimensions length*width
                    count += (length - i + 1) * (width - j + 1);
                }
            }
            let diff = Math.abs(2e6 - count);
            // closer solution found, update area of grid
            if (diff < bestDiff){
                area = length * width;
                bestDiff = diff;
            }
        }
    }
    return area;
}

console.log(euler85());

File: go/euler85.go
package main
import (
    "fmt"
    "math"
)

func main() {
    res, diff := 0, 2000000
    for length := 1; length <= 100; length++ {
        for width := 1; width <= length; width++ {
            rectangles := 0
            for i := 1; i <= length; i++ {
                for j := 1; j <= width; j++ {
                    rectangles += (length - i + 1) * (width - j + 1)
                }
            }
            newDiff := int(math.Abs(float64(2000000 - rectangles)))
            if newDiff < diff {
                diff = newDiff
                res = length * width
            }
        }
    }
    fmt.Println(res)
}
